distributed/comm/ucx.py

```diff
@@ -202,30 +202,40 @@ async def write(
                 )
                 sizes = tuple(nbytes(f) for f in frames)
 
-                host_frames_size = 0
-                device_frames_size = 0
-                for is_cuda, each_size in zip(cuda_frames, sizes):
-                    if is_cuda:
-                        device_frames_size += each_size
+                host_frames = host_array(0)
+                device_frames = device_array(0)
+                if nframes == 1:
```

Comment:
Instead of just checking `nframes == 1` I think you want to short circuit based on the number of device frames and the number of host frames individually. If we've got 2 frames here, 1 device frame and 1 host frame, each one gets copied unnecessarily.

Reply:
Woops outdated commit, disregard :)

Reply:
No worries. Yep should be the case in more recent commits.